gk/main.c

```diff
@@ -196,16 +210,17 @@ drop_packet(struct rte_mbuf *pkt)
  * (3) put this encapsulated packet in the request queue.
  */
 static int
-gk_process_request(struct flow_entry *fe, struct ipacket *packet)
+gk_process_request(struct flow_entry *fe,
+	struct ipacket *packet, struct gk_lpm *ltbl)
 {
 	int ret;
 	uint64_t now = rte_rdtsc();
 	uint8_t priority = priority_from_delta_time(now,
 			fe->u.request.last_packet_seen_at);
-
-	/* TODO The tunnel information should come from the LPM table. */
-	struct ipip_tunnel_info tunnel;
-	memset(&tunnel, 0, sizeof(tunnel));
+	struct gk_fib *fib =
+		fe->flow.proto == ETHER_TYPE_IPv4 ?
+		&ltbl->fib_tbl[fe->grantor_id] :
+		&ltbl->fib_tbl6[fe->grantor_id];
```

Comment:
This code is becoming unnecessarily convoluted. Please rename `fe->grantor_id` to `fe->grantor_fib` and have its type as `struct gk_fib *`. This way you can address the grantor FIB entry directly and can even drop parameter `ltbl`.